src/utility/Utility.cpp

```diff
@@ -114,23 +80,134 @@ std::shared_ptr<spdlog::logger> create_logger(std::string logger_name) {
+    if (list1[counter1] < list2[counter2]) {
+        result.push_back(list1[counter1]);
+        counter1++;
+    } else {
+        result.push_back(list2[counter2]);
+        counter2++;
+    }
+    while (counter1 < size_list1 && counter2 < size_list2) {
+        if (list1[counter1] == result[result.size()-1]) {
+            counter1++;
+            continue;
+        }
+        if (list2[counter2] == result[result.size()-1]) {
+            counter2++;
+            continue;
+        }
+        if (list1[counter1] < list2[counter2]) {
+            result.push_back(list1[counter1]);
+            counter1++;
+        } else {
+            result.push_back(list2[counter2]);
+            counter2++;
+        }
+    }
+    if (counter1 < size_list1) {
+        if (list1[counter1] == result[result.size()-1]) {
+            counter1++;
+        }
+        for (size_t c = counter1; c < size_list1; c++) {
+            result.push_back(list1[c]);
+        }
+    } else {
+        if (list2[counter2] == result[result.size()-1]) {
+            counter2++;
+        }
+        for (size_t c = counter2; c < size_list2; c++) {
+            result.push_back(list2[c]);
+        }
+    }
+    return result;
+}
+
+void log_minimum(Field *field, const std::string& text, const std::string& logger_name) {
+#ifndef BENCHMARKING
+    auto logger = Utility::create_logger(logger_name);
+#endif
+    auto data = field->data;
+    real minimum_inner = ULONG_LONG_MAX;
+    auto boundary = BoundaryController::getInstance();
+    size_t *innerList = boundary->get_inner_list_level_joined();
+    size_t size_innerList = boundary->get_size_inner_list();
+    for (size_t i = 0; i < size_innerList; i++) {
+        size_t idx = innerList[i];
+        real value = data[idx];
+        if (value < minimum_inner) {
+            minimum_inner = value;
+        }
+    }
+#ifndef BENCHMARKING
+    logger->info("minimum inner {}: {}", text, minimum_inner);
+#endif
+    real minimum_boundary = ULONG_LONG_MAX;
+    std::vector<size_t> indices;
+    size_t *boundaryList = boundary->get_boundary_list_level_joined();
+    size_t size_boundaryList = boundary->get_size_boundary_list();
+    for (size_t i = 0; i < size_boundaryList; i++) {
+        size_t idx = boundaryList[i];
+        real value = data[idx];
+        if (value < minimum_boundary) {
+            indices.clear();
+            indices.push_back(idx);
+            minimum_boundary = value;
+        } else if (value == minimum_boundary) {
+            indices.push_back(idx);
+        }
+    }
+    // size_t Nx = Domain::getInstance()->get_Nx();
+    // size_t Ny = Domain::getInstance()->get_Ny();
+    // std::string index;
+    // for (size_t idx: indices) {
+    //     size_t k = getCoordinateK(idx, Nx, Ny);
+    //     size_t j = getCoordinateJ(idx, Nx, Ny, k);
+    //     size_t i = getCoordinateI(idx, Nx, Ny, j, k);
+    //     index += " (" + std::to_string(i) + "|" + std::to_string(j) + "|" + std::to_string(k) + ")";
+    // }
```

Comment:
dead code

Reply:
not part of the PR anymore. but I think it was already changed